src/boundary_conditions.jl

```diff
@@ -31,46 +31,42 @@ struct CubicPeriodicBoundaryConditions{cType <: Real} <: BoundaryConditions
     L::cType
 end
 
-function apply_boundary_conditions!(ri, rj, pbc::PeriodicBoundaryConditions, R2)
-    rij = @MVector [Inf, Inf, Inf]
-    success = false
-    rij2 = 0
-
-    for dx in [0,-pbc[2],pbc[2]], dy in [0,-pbc[4],pbc[4]], dz in [0,-pbc[6],pbc[6]]
-        rij = @MVector [ri[1] - rj[1] + dx, ri[2] - rj[2] + dy, ri[3] - rj[3] + dz]
-        for x in (1, 2, 3) 
-            rij[x] -= (pbc[2x] - pbc[2x - 1]) * div(rij[x], (pbc[2x] - pbc[2x - 1]))
-        end
-        rij2 = dot(rij, rij)
-        if  rij2 < R2
-            success = true
-            break
-        end
-    end
-    return (rij, rij2, success)
+function get_interparticle_distance(ri, rj, pbc::PeriodicBoundaryConditions)
```

Comment:
so is this where all of he time is spent?

Reply:
Not after my changes, but it still takes a substantial minority of the total runtime. It was along the lines of 90% before, now it's up a third or so.